### PR TITLE
HADOOP-18110. ViewFileSystem: Add Support for Localized Trash Root

### PR DESCRIPTION
**Pull request by xinglin**
Fixes #3956

(cherry picked from commit ca8ba24051b7fca4612c9c182cb49f5183ce33ba)

### Description of PR

Cherry-pick the patch from trunk to branch-2.10, with a few other changes, to make it work for branch-2.10.

### How was this patch tested?

Manually run the new test cases from Intellij.

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  12m  7s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ branch-2.10 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  14m 16s |  |  branch-2.10 passed  |
| +1 :green_heart: |  compile  |  13m  5s |  |  branch-2.10 passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  compile  |  10m 44s |  |  branch-2.10 passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  |  branch-2.10 passed  |
| +1 :green_heart: |  mvnsite  |   1m 18s |  |  branch-2.10 passed  |
| +1 :green_heart: |  javadoc  |   1m 14s |  |  branch-2.10 passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javadoc  |   1m  3s |  |  branch-2.10 passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| -1 :x: |  spotbugs  |   2m  0s | [/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/1/artifact/out/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html) |  hadoop-common-project/hadoop-common in branch-2.10 has 2 extant spotbugs warnings.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 44s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  12m 28s |  |  the patch passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javac  |  12m 28s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  10m 46s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  javac  |  10m 46s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 42s | [/results-checkstyle-hadoop-common-project_hadoop-common.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/1/artifact/out/results-checkstyle-hadoop-common-project_hadoop-common.txt) |  hadoop-common-project/hadoop-common: The patch generated 1 new + 182 unchanged - 0 fixed = 183 total (was 182)  |
| +1 :green_heart: |  mvnsite  |   1m 17s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 13s |  |  the patch passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javadoc  |   1m  6s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  spotbugs  |   2m 10s |  |  the patch passed  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   9m 27s | [/patch-unit-hadoop-common-project_hadoop-common.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/1/artifact/out/patch-unit-hadoop-common-project_hadoop-common.txt) |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 49s |  |  The patch does not generate ASF License warnings.  |
|  |   | 102m  5s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.util.TestReadWriteDiskValidator |
|   | hadoop.fs.sftp.TestSFTPFileSystem |
|   | hadoop.io.compress.TestCompressorDecompressor |
|   | hadoop.io.compress.snappy.TestSnappyCompressorDecompressor |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3994 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux c18a4756a3fd 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-2.10 / 63ccc116c4ff9798f2227fe0d5d2b727fce3079f |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_262-b10 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/1/testReport/ |
| Max. process+thread count | 1371 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/1/console |
| versions | git=2.17.1 maven=3.6.0 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by xinglin** [comment]
these unit test failures seem to be false alarms. I am able to run and pass these tests on my laptop.

  | hadoop.util.TestReadWriteDiskValidator
  | hadoop.fs.sftp.TestSFTPFileSystem
  | hadoop.io.compress.TestCompressorDecompressor
  | hadoop.io.compress.snappy.TestSnappyCompressorDecompressor



```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.hadoop.util.TestReadWriteDiskValidator
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.278 s - in org.apache.hadoop.util.TestReadWriteDiskValidator

[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.hadoop.fs.sftp.TestSFTPFileSystem
[INFO] Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.031 s - in org.apache.hadoop.fs.sftp.TestSFTPFileSystem

[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.hadoop.io.compress.TestCompressorDecompressor
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.112 s - in org.apache.hadoop.io.compress.TestCompressorDecompressor

[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.hadoop.io.compress.snappy.TestSnappyCompressorDecompressor
[WARNING] Tests run: 14, Failures: 0, Errors: 0, Skipped: 14, Time elapsed: 0.085 s
```

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 56s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ branch-2.10 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  14m 12s |  |  branch-2.10 passed  |
| +1 :green_heart: |  compile  |  13m  4s |  |  branch-2.10 passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  compile  |  10m 53s |  |  branch-2.10 passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 43s |  |  branch-2.10 passed  |
| +1 :green_heart: |  mvnsite  |   1m 18s |  |  branch-2.10 passed  |
| +1 :green_heart: |  javadoc  |   1m 16s |  |  branch-2.10 passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javadoc  |   1m  5s |  |  branch-2.10 passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| -1 :x: |  spotbugs  |   2m  1s | [/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/2/artifact/out/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html) |  hadoop-common-project/hadoop-common in branch-2.10 has 2 extant spotbugs warnings.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 46s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  12m 27s |  |  the patch passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javac  |  12m 27s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  10m 49s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  javac  |  10m 49s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 18s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 13s |  |  the patch passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javadoc  |   1m  5s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  spotbugs  |   2m 10s |  |  the patch passed  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   9m 25s | [/patch-unit-hadoop-common-project_hadoop-common.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/2/artifact/out/patch-unit-hadoop-common-project_hadoop-common.txt) |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 49s |  |  The patch does not generate ASF License warnings.  |
|  |   |  91m 13s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.util.TestReadWriteDiskValidator |
|   | hadoop.util.TestDiskCheckerWithDiskIo |
|   | hadoop.fs.sftp.TestSFTPFileSystem |
|   | hadoop.io.compress.TestCompressorDecompressor |
|   | hadoop.io.compress.snappy.TestSnappyCompressorDecompressor |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3994 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 74522267b2e6 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-2.10 / b0c54fa62d91748706a5c22cb961926d98bf580c |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_262-b10 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/2/testReport/ |
| Max. process+thread count | 1368 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/2/console |
| versions | git=2.17.1 maven=3.6.0 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ConfigUtil.java [hunk 1]
```diff
@@ -67,7 +67,28 @@ public static void addLink(final Configuration conf, final String src,
...
   }
-  
+
+  /**
+   * Add a LinkFallback to the config for the specified mount table.
+   * @param conf
+   * @param mountTableName
+   * @param target
+   */
+  public static void addLinkFallback(Configuration conf,
```
Comment (omalley) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ConfigUtil.java]: This change is out of scope for the backport.

**Review by omalley — CHANGES_REQUESTED** [review]

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java [hunk 2]
```diff
@@ -3627,11 +3627,12 @@ void decrUserConnections(String user) {
           return;
         } else {
           count--;
-        }
-        if (count == 0) {
-          userToConnectionsMap.remove(user);
-        } else {
-          userToConnectionsMap.put(user, count);
+
+          if (count == 0) {
```
Comment (omalley) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java]: This looks like a spurious change also.

Comment (xinglin) [reply hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java]: removed.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 55s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  1s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ branch-2.10 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  14m 17s |  |  branch-2.10 passed  |
| +1 :green_heart: |  compile  |  13m 29s |  |  branch-2.10 passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  compile  |  11m 27s |  |  branch-2.10 passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 37s |  |  branch-2.10 passed  |
| +1 :green_heart: |  mvnsite  |   1m 16s |  |  branch-2.10 passed  |
| +1 :green_heart: |  javadoc  |   1m  9s |  |  branch-2.10 passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javadoc  |   0m 59s |  |  branch-2.10 passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| -1 :x: |  spotbugs  |   2m  3s | [/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/3/artifact/out/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html) |  hadoop-common-project/hadoop-common in branch-2.10 has 2 extant spotbugs warnings.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 45s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  13m  2s |  |  the patch passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javac  |  13m  2s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  11m 42s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  javac  |  11m 42s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 38s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 12s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 10s |  |  the patch passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javadoc  |   1m  1s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  spotbugs  |   2m 12s |  |  the patch passed  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   9m 54s | [/patch-unit-hadoop-common-project_hadoop-common.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/3/artifact/out/patch-unit-hadoop-common-project_hadoop-common.txt) |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 46s |  |  The patch does not generate ASF License warnings.  |
|  |   |  93m  1s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.util.TestReadWriteDiskValidator |
|   | hadoop.fs.sftp.TestSFTPFileSystem |
|   | hadoop.io.compress.TestCompressorDecompressor |
|   | hadoop.io.compress.snappy.TestSnappyCompressorDecompressor |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3994 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 1da12856682e 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-2.10 / 715defe02a454c3de62b72bc511203355ba79b0d |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_262-b10 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/3/testReport/ |
| Max. process+thread count | 1366 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/3/console |
| versions | git=2.17.1 maven=3.6.0 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m 15s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ branch-2.10 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  15m  1s |  |  branch-2.10 passed  |
| +1 :green_heart: |  compile  |  13m 47s |  |  branch-2.10 passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  compile  |  11m 16s |  |  branch-2.10 passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 39s |  |  branch-2.10 passed  |
| +1 :green_heart: |  mvnsite  |   1m 17s |  |  branch-2.10 passed  |
| +1 :green_heart: |  javadoc  |   1m 17s |  |  branch-2.10 passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javadoc  |   1m  3s |  |  branch-2.10 passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| -1 :x: |  spotbugs  |   2m  3s | [/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/4/artifact/out/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html) |  hadoop-common-project/hadoop-common in branch-2.10 has 2 extant spotbugs warnings.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 48s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  13m 36s |  |  the patch passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javac  |  13m 36s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  11m 33s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  javac  |  11m 33s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 43s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 16s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 14s |  |  the patch passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javadoc  |   1m  5s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  spotbugs  |   2m 11s |  |  the patch passed  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   9m 25s | [/patch-unit-hadoop-common-project_hadoop-common.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/4/artifact/out/patch-unit-hadoop-common-project_hadoop-common.txt) |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 48s |  |  The patch does not generate ASF License warnings.  |
|  |   |  95m 14s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.util.TestReadWriteDiskValidator |
|   | hadoop.fs.sftp.TestSFTPFileSystem |
|   | hadoop.io.compress.TestCompressorDecompressor |
|   | hadoop.io.compress.snappy.TestSnappyCompressorDecompressor |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3994 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 7300a36b11cc 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-2.10 / dfd58e32183729f14b08edff32c4c77d89acc4a0 |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_262-b10 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/4/testReport/ |
| Max. process+thread count | 1373 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/4/console |
| versions | git=2.17.1 maven=3.6.0 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  11m 35s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ branch-2.10 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  14m 36s |  |  branch-2.10 passed  |
| +1 :green_heart: |  compile  |  14m 11s |  |  branch-2.10 passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  compile  |  11m 28s |  |  branch-2.10 passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 36s |  |  branch-2.10 passed  |
| +1 :green_heart: |  mvnsite  |   1m 16s |  |  branch-2.10 passed  |
| +1 :green_heart: |  javadoc  |   1m 14s |  |  branch-2.10 passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javadoc  |   0m 59s |  |  branch-2.10 passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| -1 :x: |  spotbugs  |   2m  2s | [/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/5/artifact/out/branch-spotbugs-hadoop-common-project_hadoop-common-warnings.html) |  hadoop-common-project/hadoop-common in branch-2.10 has 2 extant spotbugs warnings.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 46s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  13m 34s |  |  the patch passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javac  |  13m 34s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  11m 26s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  javac  |  11m 26s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 38s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 15s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 12s |  |  the patch passed with JDK Azul Systems, Inc.-1.7.0_262-b10  |
| +1 :green_heart: |  javadoc  |   1m  0s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07  |
| +1 :green_heart: |  spotbugs  |   2m 12s |  |  the patch passed  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |  10m 52s | [/patch-unit-hadoop-common-project_hadoop-common.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/5/artifact/out/patch-unit-hadoop-common-project_hadoop-common.txt) |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 46s |  |  The patch does not generate ASF License warnings.  |
|  |   | 106m  8s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.fs.sftp.TestSFTPFileSystem |
|   | hadoop.io.compress.TestCompressorDecompressor |
|   | hadoop.io.compress.snappy.TestSnappyCompressorDecompressor |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3994 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux ccf645716115 4.15.0-163-generic #171-Ubuntu SMP Fri Nov 5 11:55:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-2.10 / dfd58e32183729f14b08edff32c4c77d89acc4a0 |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_262-b10 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~18.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/5/testReport/ |
| Max. process+thread count | 1463 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3994/5/console |
| versions | git=2.17.1 maven=3.6.0 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Review by omalley — APPROVED** [review]

**Comment by xinglin** [comment]
merged into branch-2.10 by Owen. Thank you!
